### ISSUE TITLE
`RLHFDataset` accepts a prompt column that matches nothing and prompts that tokenize to zero tokens

### ISSUE THREAD

**Issue by CodersAcademy006** [ellanorai/ThinkRL#77]
Two validation gaps in `thinkrl/data/datasets.py` both end in a run that looks successful and trains on nothing or on noise.

**A prompt column that matches no row produces an empty dataset and only a warning.**

`datasets.py:145-149` logs a warning and continues when every row has been filtered out. Combined with `cli/grpo.py:249`, an empty dataset gives `total_steps = epochs * 0 // batch_size = 0`, which is the condition described in #64.

```
dataset has 16 rows, column is 'question'
WARNING datasets.py:146] No valid samples found in ... with prompt_column='prompt'. Dataset length: 16
len(dataset) = 0
cli/grpo.py:249  total_steps = 1 * 0 // 4 = 0
```

This is reachable through the CLI defaults. `--prompt-column` defaults to `"prompt"` at `cli/grpo.py:34`, and the help text for `--dataset-config` at `cli/grpo.py:32` suggests gsm8k, whose columns are `question` and `answer`. So `thinkrl grpo --model X --ref-model X --dataset gsm8k --dataset-config main` loads the dataset, keeps zero rows, runs zero steps and reports completion.

**A prompt that tokenizes to zero tokens becomes a row of pure padding.**

`datasets.py:131-133` validates that the prompt string is non-empty, but nothing checks the token count after `datasets.py:167-174` tokenizes it. A prompt the tokenizer reduces to nothing survives, collates into an all-padding row, and `GRPOTrainer.make_experience` calls `generate()` on it with no guard:

```
dataset kept 16 rows; rows tokenizing to ZERO tokens: [8]
  row 8: prompt_text='what is 9 plus 9' -> input_ids shape (0,)
generate() calls: 4   rows passed in with attention_mask.sum()==0: 1
  call 3 row 3: input_ids=[1, 1, 1, 1] attention_mask=[0, 0, 0, 0]
```

The model generates from an empty context, the completion is scored by the reward function, and it enters its GRPO group's mean and standard deviation. It therefore corrupts the advantages of the real completions grouped with it, not only its own. Real triggers include prompts entirely out of vocabulary and text in a script the tokenizer drops. The only visible symptom is a transformers warning about right-padding, which points at the wrong thing: the padding side is correct, and the row is entirely padding because the prompt vanished.

**Suggested fix**

Raise rather than warn when the dataset is empty after filtering, naming the requested column and the columns actually present. Drop rows that tokenize to zero tokens and log the count. Optionally assert `attention_mask.sum(dim=1) > 0` before generation as a second guard.

I am happy to open a PR for this.
